src/connector/src/common.rs

```diff
@@ -51,6 +50,111 @@ pub struct AwsPrivateLinkItem {
     pub port: u16,
 }
 
+use aws_config::default_provider::region::DefaultRegionChain;
+use aws_config::sts::AssumeRoleProvider;
+use aws_credential_types::provider::SharedCredentialsProvider;
+use aws_types::region::Region;
+use aws_types::SdkConfig;
+
+/// A flatten cofig map for aws auth.
+#[derive(Deserialize, Serialize, Debug, Clone, WithOptions)]
+pub struct AwsAuthProps {
+    pub region: Option<String>,
+    #[serde(alias = "endpoint_url")]
+    pub endpoint: Option<String>,
+    pub access_key: Option<String>,
+    pub secret_key: Option<String>,
+    pub session_token: Option<String>,
+    pub arn: Option<String>,
+    pub external_id: Option<String>,
+    pub profile: Option<String>,
+}
+
+impl Default for AwsAuthProps {
+    fn default() -> Self {
+        Self {
+            region: Default::default(),
+            endpoint: Default::default(),
+            access_key: Default::default(),
+            secret_key: Default::default(),
+            session_token: Default::default(),
+            arn: Default::default(),
+            external_id: Default::default(),
+            profile: Some("default".to_string()),
```

Comment:
Should we also make it a `#[serde(default)]`? Also they can be unified with `serde_default`.

Reply:
Maybe we should remove the `"default"`, as there are usages like `profile: Default::default(),`....

Reply:
<img width="612" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/37948597/67a3b93a-32a6-4368-9fec-d0de7d360724">